device/sycl/src/utils/thread_id.hpp

```diff
@@ -0,0 +1,67 @@
+/**
+ * traccc library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/device/concepts/thread_id.hpp"
+
+// SYCL include(s).
+#include <sycl/sycl.hpp>
+
+namespace traccc::sycl::details {
+
+/// A SYCL thread identifier type
+template <std::size_t DIMENSIONS = 1>
```

Comment:
What's the proposed strategy for extending this to 2 and 3 dimensions? Wouldn't it be nicer to declare this as an empty type and then specialize it for $D \in \{1, 2, 3\}$?

Reply:
Not sure. :thinking: But I wouldn't jump to template specialization with this. That just has so much code repetition built in...

Reply:
Thinking about it, we can simply declare them as:

```
getLocalThreadIdY() requires(DIMENSIONS >= 2) {
...
}
```

So I think that's probably a fine solution. :+1: